packages/core/src/utils/QueryHelper.ts

```diff
@@ -95,7 +95,11 @@ export class QueryHelper {
       QueryHelper.inlinePrimaryKeyObjects(where as Dictionary, meta, metadata);
     }
 
-    where = QueryHelper.processParams(where) || {};
+    where = QueryHelper.processParams(where);
+
+    if (!where && typeof where !== 'number') {
```

Comment:
this should do the same, right?
```suggestion
    if (where == null) {
```

Reply:
I think empty string as pk should still return `{}`, so not same for that case

Reply:
rest of the codebase is using the `null` checks, so this part should do it the same way. i disagree that an empty string should be treated differently than `0`

Reply:
Ok will need to update other test that fails with that change

Reply:
Could do `QueryHelper.processParams(where) ?? {}` then

Reply:
yeah correct, lets do it that way